middleware/expandDocument.js

```diff
@@ -0,0 +1,72 @@
+const dotProp = require('dot-prop');
+const _ = require('lodash');
+
+function service(esclient) {
+  function expandDocument(doc) {
+    const traverse = (obj, cb, keypath) => {
```

Comment:
It's a convention in js/node to have the callback as the final argument, is it possible to change the function signature to adopt this convention?

Reply:
Could you please add a couple of comment lines about what `traverse` does?

Reply:
added.

it wasn't the final argument because it's optional because it's a recursive/accumulator style function. changed to use a helper.

Reply:
It doesn't look like these functions need to be redeclared on each invocation of the parent?

Can you please move the functions up to the module scope so that `traverse` can be unit tested?